btk/measure.py

```diff
@@ -1,175 +1,239 @@
-
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                         images, isolated images, observing conditions and
-                         blend catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                         measurement on.
-
-        Returns:
-            output of measurement algorithm (fluxes, shapes, size, etc.) as
-            an astropy catalog.
-        """
-        return None
-
-    def get_deblended_images(self, data, index):
-        """Function describing how the deblending algorithm is run.
-
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                         images, isolated images, observing conditions and
-                         blend catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                         measurement on.
-
-        Returns:
-            output of deblending algorithm as a dict.
-        """
-        return None
-
+def basic_measure(batch, idx):
+    """Return centers detected when object detection is performed on the
+    input image with skimage.feature.peak_local_max.
+
+    NOTE: This function does not support the multi-resolution feature.
+
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
+
+    Returns:
+            dict containing catalog with entries corresponding to measured peaks.
+    """
+    if isinstance(batch["blend_images"], dict):
+        raise NotImplementedError("This function does not support the multi-resolution feature.")
+
+    image = np.mean(batch["blend_images"][idx], axis=0)
+
+    # set detection threshold to 5 times std of image
+    threshold = 5 * np.std(image)
+    coordinates = peak_local_max(image, min_distance=2, threshold_abs=threshold)
+
+    # construct catalog from measurement.
+    catalog = astropy.table.Table()
+    catalog["dx"] = coordinates[:, 1]
+    catalog["dy"] = coordinates[:, 0]
+    return {"catalog": catalog}
+
+
+def sep_measure(batch, idx):
+    """Return centers detected when object detection and photometry
+    is done on input image with SEP.
+
+    NOTE: This function does not support the multi-resolution feature.
+
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
+
+    Returns:
+        dict with the centers of sources detected by SEP detection algorithm.
+    """
+    if isinstance(batch["blend_images"], dict):
+        raise NotImplementedError("This function does not support the multi-resolution feature.")
+
+    image = batch["blend_images"][idx]
+    stamp_size = image.shape[-2]  # true for both 'NCHW' or 'NHWC' formats.
```

Comment:
This needs a longer comment, since it can lead to errors down the line. 
One should probably add in the method docstring that the blend images are ndarray expected in a certain format.

Reply:
This should not be necessary since the images are supposed to come from a DrawBlendsGenerator directly, meaning they will be in the right format. When we make a tutorial for this, we should specify that images may come in the two different formats if users want to add their own measure_function.